client/src/hooks/useVersion.ts

```diff
@@ -0,0 +1,19 @@
+import { useBooleanToggle } from '@mantine/hooks';
+import { useEffect, useState } from 'react';
+import getVersion from '../api/getVersion';
+
+const useVersion = () => {
+	const [version, setVersion] = useState<string>('');
+	const [fetching, setFetching] = useBooleanToggle(false);
+
+	useEffect(() => {
```

Comment:
This should be a useCallback instead and be returned as part of the below array as a "refetch" function. The useEffect can call that function on mount.